Unapproved unread

emanuele45 opened the pull request:
Sending it to keep track, useful but not critical I think (due to the queries involved)

emanuele45 committed:
Unapproved posts show up in unread and unreadreplies [Bug 4426] - separated branch because it can introduce security issues, so I prefer to test it a little bit more

Signed-off-by: emanuele <emanuele45@gmail.com>

emanuele45 committed:
Users should be able to see their posts even if they did them in a board that now is no more accessible (at least in their profile) [msg 3256225 and 3256555]

Signed-off-by: emanuele <emanuele45@gmail.com>

live627 commented on Sources/Recent.php:
```diff
@@ -21,6 +21,7 @@
  *
  * - respects approved, recycled, and board permissions
  * - @todo is this even used anywhere?
+ * - emanuele: apparently no...
```
If this function is unused, consider removing it, perhaps?

emanuele45 replied:
And spoil all the fun? :P

live627 commented:
> useful but not critical I think (due to the queries involved)

Did they really make that much of a difference?

emanuele45 commented:
> Did they really make that much of a difference?

No idea if you speak about performance. My comment (indeed a bit cryptic) was more about the fact that a bit of testing for privacy-related matters (i.e. be sure that people do not access things they shouldn't) is good and needed.

...why your comment is up there and mine is down here? ??? ...lol

live627 commented:
Comments to the pull request are shown in chronological order. I commented in the difff after I commented on the PR. Yours came last, so it shows last.

About the query comment - I thought you meant performance. But privacy, yes, it is a big concern. On that subject, I'd think the changes to query_see_board to allow the owner to view all their posts regardless isn't such a good idea. I see two reasons against this: 1, their post in the profile is of course out of context, and when they click to view the topic, they are denied, a probable support hassle; and 2, the hidden  post might be a sooper-seekrit discussion related to staff duties that they may not view because they were demoted.

emanuele45 commented:
Unfortunately I thought I explained the reason of the change in the commit message, but I didn't, I just recorded the messages id that are not in public boards so are useless...sorry.

This feature was requested by Kindred and Lex because it appears it could be/become (not clear to me at the moment if it is, is not, it could be, etc.) a legal requirement in Canada and Finland. But since we are already (not) dealing (well, in fact with 2.1 we did help in dealing with it providing a central function to set cookies) with other "not-so-clear" legal requirements (UK and ICO on cookies) that's another reason the PR is not critical.

I like the feature because I think it's how it should work (it is also a way to control if my posts have been changed after the moment I stopped having access to a certain area, from time to time I can be paranoid :P), there was a request, so I did it. Thought I don't have a strong opinion.

That said, I can understand the "support hassle" (even though I'm not particularly concerned :P).
The fact that the post is out of context yes, although I see good reasons I want to be able to see all the posts I did (stupid example: a note about a bug I wrote somewhere I don't have access any more).
About point two IMHO is completely useless to worry about it: as soon as I can access a page I can configure my browser to automatically save it (like I set any of my IM clients to keep logs of any conversation I do) and have permanent access to any information (not only my posts) I had access while in any kind of sooper-seekrit area.

jdarwood007 commented:
FYI, handle_unapprove is not a good variable name.  Using SMFs naming context it should be can_see_unapproved.

I am not so sure about the second commit.  Some people may see that as a security issue rather than as the feature it is intended to be.  Perhaps a option here is better suited to allow proper customization/security setup?

emanuele45 committed:
Revert "Users should be able to see their posts even if they did them in a board that now is no more accessible (at least in their profile) [msg 3256225 and 3256555]"

This reverts commit 933279cd5315daf0771fc87084534af07ee93ec7.

emanuele45 committed:
Changed variable name

Signed-off-by: emanuele <emanuele45@gmail.com>

emanuele45 committed:
action recent should respect unapproved topics (thanks Arantor) [Topic 482262] and show unapproved things to the moderators

Signed-off-by: emanuele <emanuele45@gmail.com>

emanuele45 committed:
Merge branch 'master' into unapproved_unread

emanuele45 committed:
That should work better than before in showing the unapproved messages

Signed-off-by: emanuele <emanuele45@gmail.com>

emanuele45 commented:
Okay, reverted the "new" feature, kept the bug-fix (that was the first commit) and added another one related to recent messages.
I hope nothing is too wrong... :P

emanuele45 committed:
Merge branch 'master' into unapproved_unread

emanuele45 committed:
Yet another version that should finally work properly...probably more a mod though at that point

Signed-off-by: emanuele <emanuele45@gmail.com>

emanuele45 commented:
I don't trust it, better as a mod for the next release.

norv commented:
There are other issues I've discussed at some point with @Spuds on this. While I like the feature (a lot), and I think we should make it default or possible behavior in some way, I don't think it's appropriate for 2.1. Random example of issues: a post intended to serve as collection of personal information, which keeps being modified by others, after you cannot access it. Nobody expects the _modified_ post information to be available to you, given the current behavior of SMF, and changing it requires more thought, deeper code changes, documentation and feedback from the community.

There are other issues as well.
